src/main/java/com/lgtm/easymoney/services/impl/SearchServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+        var userWrapper = userRepository.findById(id);
+        if (userWrapper.isEmpty()) {
+            throw new ResourceNotFoundException("User", "id", id);
+        }
+        
+        return userWrapper.get();
+    }
+
+    @Override
+    public List<User> getUserByName(String accountName) {
+        var accountWrapper = accountRepository.findByAccountNameContainingIgnoreCase(accountName);
+        if (accountWrapper.isEmpty()) {
+            throw new ResourceNotFoundException("Account", "accountName", accountName);
+        }
+        
+        List<User> userList = new ArrayList<User>();
+        for(Account account : accountWrapper){
+            userList.add(account.getAccountUser());
+        }
+        
+        return userList;
+    }
+
+    @Override
+    public List<User> getUserByEmailOrPhone(String email, String phone) {
+        var userList = userRepository.findByEmailContainingIgnoreCaseOrPhoneContaining(email, phone);
+        if (userList.isEmpty()) {
+            //TODO: better exception handling to verify which field is empty
+            throw new ResourceNotFoundException("User", "email", email);
+        }
+        return userList;
+    }
+
+    @Override
+    public ResponseEntity<SearchRsp> searchByID(Long id) {
+        //Getting user by id
+        User user = getUserByID(id);
+        boolean searchResult = true;
+ 
+        //Compose response
+        List<ProfileRsp> profileList = new ArrayList<ProfileRsp>();
+        ProfileRsp res = new ProfileRsp();
+        res.setAccountName(user.getAccount().getAccountName());
+        res.setEmail(user.getEmail());
+        res.setAddress(user.getAddress());
+        res.setUserType(user.getType());
+        res.setPhone(user.getPhone());
+        profileList.add(res);
+        SearchRsp searchRes = new SearchRsp();
+        searchRes.setSuccess(searchResult);
+        searchRes.setUserProfiles(profileList);
+        
+        //Issue response
+        if (searchResult) {
+            return ResponseEntity.status(HttpStatus.OK).body(searchRes);
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(searchRes);
+    }
+
+    @Override
+    public ResponseEntity<SearchRsp> searchByInfo(String userInfo) {
+        
+        List<User> userList;
+        
+        //regex to check valid phone number
+        String phoneRegex = "^(?:(?:\\+?1\\s*(?:[.-]\\s*)?)?(?:\\(\\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\\s*\\)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\\s*(?:[.-]\\s*)?)?([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\\s*(?:[.-]\\s*)?([0-9]{4})(?:\\s*(?:#|x\\.?|ext\\.?|extension)\\s*(\\d+))?$";
+        
+        //regex to check valid email pattern
+        Pattern p = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+        java.util.regex.Matcher matcher = p.matcher(userInfo);
+
+        //Parse userInfo
+        if(matcher.find() || userInfo.matches(phoneRegex)) {
```

Comment:
What if the client wants list of accountname matched union list of email matched?